main/utils/macros.h

```diff
@@ -51,7 +45,26 @@
  * @param arr An array to find the number of elements in
  * @return The the number of elements in an array (not the byte size!)
  */
-#define ARRAY_SIZE(arr) (sizeof(arr) / sizeof((arr)[0]) + STATIC_EXP(IS_ARRAY(arr)))
+#ifdef __APPLE__
+    /* Force a compilation error if condition is true, but also produce a
+       result (of value 0 and type size_t), so the expression can be used
+       e.g. in a structure initializer (or where-ever else comma expressions
+       aren't permitted). */
+    #define BUILD_BUG_ON_ZERO(e) (sizeof(struct { int : -!!(e); }))
```

Comment:
Can someone make sure this is okay by the way? It's [coming from Linux directly](https://github.com/torvalds/linux/blob/6f0d349d922ba44e4348a17a78ea51b7135965b1/tools/include/linux/bug.h), so I'd imagine it's fine, but it is licensed under `GPL-2.0` according to the Linux source code, so I want to make sure there's no issues here.

Reply:
I'm not a legal eagle, but everything we do is publically available, so it seems fine to use GPL2 code, especially a snippet like this.